src/atmega328p/com/i2c.rs

```diff
@@ -0,0 +1,370 @@
+    }
+
+    /// Sends a Repeat Start Signal
+    /// Usage: rustduino::atmega328p::com::i2c::rep_start()
+    pub fn rep_start(&mut self) -> bool {
+        self.twcr.write(0xA4); // TWINT TWSTA and TWA set to 1
+
+        return self.wait_to_complete(REP_START);
+    }
+
+    /// Stops the TWI bus.
+    /// Usage: rustduino::atmega328p::com::i2c::stop()
+    pub fn stop(&mut self) {
+        self.twcr.write(0xB0);
+    }
+
+    /// Sets address of Slave.
+    /// Used in Master Slave Mode.
+    /// Returns true if process is successful
+    /// Usage: rustduino::atmega328p::com::i2c::set_addr(address:u8)
+    pub fn set_address(&mut self, addr: u8) -> bool {
+        self.twdr.write(addr << 1 & !0x01); // loading SLA_W to TWDR
+        self.twcr.update(|cr| {
+            cr.set_bit(TWINT, true);
+            cr.set_bit(TWEN, true);
+        });
+
+        return self.wait_to_complete(MT_SLA_ACK);
+    }
+
+    /// Checks if slave is acknowledged.
+    /// Returns true if process is successful
+    /// Usage: rustduino::atmega328p::com::i2c::address_read(address:u8)
+    pub fn address_read(&mut self, address: u8) -> bool {
+        self.twdr.write(address << 1 | 0x01);
+        self.twcr.update(|cr| {
+            cr.set_bit(TWINT, true);
+            cr.set_bit(TWEN, true);
+        });
+
+        return self.wait_to_complete(MR_SLA_ACK);
+    }
+
+    /// Writes one byte of data to the Slave.
+    /// Need to set address first.
+    /// Returns true if process is successful
+    /// Usage: rustduino::atmega328p::com::i2c::write(data:u8)
+    pub fn write(&mut self, data: u8) -> bool {
+        delay_ms(1);
+        // write_sda(); // doubt if neended
+        self.twdr.write(data);
+        self.twcr.write(0x84); // TWCR = (1<<TWINT)|(1<<TWEN);
+
+        return self.wait_to_complete(MT_DATA_ACK);
+    }
+
+    /// Writes consecutive bytes of data to the Slave.
+    /// Need to set address first.
+    /// Returns number of bytes written
+    /// Usage: rustduino::atmega328p::com::i2c::write_burst(data:&FixedSliceVec<u8>)
+    pub fn write_burst(&mut self, data: &FixedSliceVec<u8>) -> usize {
+        let mut x: usize = 0;
+        while x < data.len() {
+            delay_ms(1);
+            if !self.write(data[x]) {
+                break;
+            }
+            x += 1;
+        }
+        return x + 1;
+    }
+
+    /// Appends the value in TWCR to the given vector.
+    /// Need to set address first.
+    /// Returns true if process is completed.
+    /// Usage: rustduino::atmega328p::com::i2c::read_ack(data:&FixedSliceVec<u8>)
+    pub fn read_ack(&mut self, data: &mut FixedSliceVec<u8>) -> bool {
+        self.twcr.write(0xC4); //TWCR = (1 << TWINT) | (1 << TWEA) | (1 << TWEN)
+        data.push(self.twdr.read());
+
+        return self.wait_to_complete(MR_DATA_ACK);
+    }
+
+    /// Appends the value in TWCR to the given vector.
+    /// Need to set address first.
+    /// Returns true if process is completed.
```

Comment:
Please don't write comments as points, unless they are bullet points, change it to proper sentences, since formatting would be removed,